tools/idf.py

```diff
@@ -713,7 +714,54 @@ def main() -> None:
         else:
             raise
     else:
-        cli(sys.argv[1:], prog_name=PROG, complete_var=SHELL_COMPLETE_VAR)
+        argv = expand_file_arguments(argv or sys.argv[1:])
+
+        cli(argv, prog_name=PROG, complete_var=SHELL_COMPLETE_VAR)
+
+
+def expand_file_arguments(argv):
+    """
+    Any argument starting with "@" gets replaced with all values read from a text file.
+    Text file arguments can be split by newline or by space.
+    Values are added "as-is", as if they were specified in this order
+    on the command line.
+    """
+    visited = set()
+    expanded = False
+
+    def expand_args(args, parent_path, file_stack):
+        expanded_args = []
+        for arg in args:
+            if not arg.startswith("@"):
+                expanded_args.append(arg)
+            else:
+                nonlocal expanded, visited
+                expanded = True
+
+                file_name = arg[1:]
+                rel_path = os.path.normpath(os.path.join(parent_path, file_name))
+
+                if rel_path in visited:
+                    file_stack_str = " -> ".join(["@" + f for f in file_stack + [file_name]])
+                    raise FatalError(f"Circular dependency in file argument expansion: {file_stack_str}")
+                visited.add(rel_path)
+
+                try:
+                    with open(rel_path, "r") as f:
+                        for line in f:
+                            expanded_args.extend(expand_args(shlex.split(line), os.path.dirname(rel_path), file_stack + [file_name]))
+                except IOError:
+                    file_stack_str = " -> ".join(["@" + f for f in file_stack + [file_name]])
```

Comment:
Please rebase to the current master branch. The fix for the failing pipeline is present there.

The pre-commit will maybe raise some errors that should be fixed. One of them will be to change the double quotations `"` for single quotations `'` in ifd.py where possible.
```suggestion
                    file_stack_str = ' -> '.join(['@' + f for f in file_stack + [file_name]])
```